src/documentBuilder.ts

```diff
@@ -255,11 +257,19 @@ export class DocumentBuilder {
       ...this.marshalMetadata(),
       ...this.marshalCompressedBinaryData(),
       ...this.marshalPermissions(),
-      documentId: uri,
+      ...this.marshalDocumentId(),
     };
     return out;
   }
 
+  private marshalDocumentId() {
+    return {
+      documentId: z.string().url().safeParse(this.uri).success
+        ? this.uri
+        : pathToFileURL(this.uri).toString(),
```

Comment:
mmh I don't know if this is a good idea. Documents might be indexed with user info in the path

Reply:
Throwing an error maybe? I don't know if generating an uri without the user knowing it is a good idea.

Reply:
So if a user index local content I should ask him to provide absolute path structured with `file://` ?
I think that doesn't work in terms of UX chief :P